packages/sdk/src/core/teamsfx.browser.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TokenCredential } from "@azure/identity";
+import { TeamsUserCredential } from "../credential/teamsUserCredential.browser";
+import { IdentityType } from "../models/identityType";
+import { UserInfo } from "../models/userinfo";
+import { formatString } from "../util/utils";
+import { ErrorWithCode, ErrorCode, ErrorMessage } from "../core/errors";
+import { internalLogger } from "../util/logger";
+import { TeamsFxConfiguration } from "../models/teamsfxConfiguration";
+
+/**
+ * A class providing credential and configuration.
+ * @beta
+ */
+export class TeamsFx implements TeamsFxConfiguration {
+  private configuration: Map<string, string | undefined>;
+  private teamsUserCredential?: TeamsUserCredential;
+  public identityType: IdentityType;
+
+  constructor(identityType?: IdentityType) {
+    this.identityType = identityType ?? IdentityType.User;
+    if (this.identityType !== IdentityType.User) {
+      const errorMsg = formatString(
+        ErrorMessage.IdentityTypeNotSupported,
+        this.identityType.toString(),
+        "TeamsFx"
+      );
+      internalLogger.error(errorMsg);
+      throw new ErrorWithCode(errorMsg, ErrorCode.IdentityTypeNotSupported);
+    }
+    this.configuration = new Map<string, string>();
+    this.loadFromEnv();
+  }
+
+  private loadFromEnv(): void {
+    let env: any;
+    if (window && (window as any).__env__) {
+      // testing purpose
+      env = (window as any).__env__;
+    } else {
+      env = process.env;
```

Comment:
Did this work in pure browser environment? Because there is no process object

Reply:
No, I think you mean the pure browser is importing SDK by <script> tag. This approach is not supported now so I assume SDK has the dependency of webpack to inject `process` object.